src/deepcode/lib/modules/LoginModule.ts

```diff
@@ -23,31 +26,38 @@ abstract class LoginModule extends ReportModule implements DeepCode.LoginModuleI
     try {
       const checkCurrentToken = await this.checkSession();
       if (checkCurrentToken) return;
-      const result = await http.login(this.baseURL, this.source);
-      const { sessionToken, loginURL } = result;
-      if (!sessionToken || !loginURL) {
+      const response = await startSession({ baseURL: this.baseURL, source: this.source });
```

Comment:
This new flow breaks a fundamental user interaction:
1. The user clicks on the login button
2. The browser opens for the login (the token is not recorded yet)
3. The user takes more than 20 seconds to onboard and login (the waitLoginConfirmation exits with no success)
4. The user is told on the browser to close the tab and go back to VSC
5. The user still sees the login button

-> by clicking on it, in the previous flow would just check the token and move to the accept-upload step. Now however the flow goes back to point 2, creating a new token and everything else. The previously created token is lost.

Reply:
Yes, this is possible, but still i don't see here a better way to handle login process.
Previously, we were saving a token to the extension, that is not confirmed. This caused many problems.
I believe, we have to save only confirmed tokens. It also simplifies following logics, when we expect token to be confirmed when we use it, with just handling an error if it's not.

Reply:
done